plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -642,21 +648,324 @@ Invoker createInvoker() {
         return LoggingInvoker.create(isSandbox());
     }
 
+    protected static Throwable reportSuspectedMethodTooLarge(Throwable x) {
+        // Suspected groovyjarjarasm.asm.MethodTooLargeException or a
+        // org.codehaus.groovy.control.MultipleCompilationErrorsException
+        // whose collection of errors refers to MethodTooLargeException.
+        // Per review comments, we do not want to statically compile a
+        // dependency on the groovyjarjarasm.asm.MethodTooLargeException
+        // internals, so gauge hitting it via String name comparisons.
+        // Other cases may be (subclasses of) RuntimeException or Error.
+        // Note that all of MultipleCompilationErrorsException, and
+        // MethodTooLargeException and CpsCompilationErrorsException
+        // are descended from RuntimeException.
+        Throwable mtlEx = null;
+        int ecCount = 0;
+        String xStr = Functions.printThrowable(x); // includes x.getMessage() contents
+        final Pattern LINE_SEP_PATTERN = Pattern.compile("\\R");
+        String[] xLines = LINE_SEP_PATTERN.split(xStr);
+
+        if (x.getClass().getSimpleName().equals("MethodTooLargeException")) {
+            mtlEx = x;
+            ecCount = 1;
+        } else if (x instanceof MultipleCompilationErrorsException) {
+            ErrorCollector ec = ((MultipleCompilationErrorsException) x).getErrorCollector();
+            ecCount = ec.getErrorCount();
+
+            for (int i = 0; i < ecCount; i++) {
+                Exception ex = ec.getException(i);
+                if (ex == null)
+                    continue;
+
+                METHOD_TOO_LARGE_LOGGER.log(Level.FINE,
+                        "CpsFlowExecution.reportSuspectedMethodTooLarge: " +
+                        "Collected Exception #" + i + ": " + ex.toString());
+                if (ex.getClass().getSimpleName().equals("MethodTooLargeException")) {
+                    mtlEx = ex;
+                    break;
+                }
+            }
+        } else if (x instanceof CpsCompilationErrorsException) {
+            // Defined in this plugin, to clone a message and stack trace
+            // from a MultipleCompilationErrorsException and be serializable.
+            // Grep it as text for "MethodTooLargeException" and "1 error"
```

Comment:
Would it not make more sense to just enrich `CpsCompilationErrorsException` with whatever information you need? AFAICT all you are looking for ultimately is the `MethodTooLargeException.message`. So, we could just keep that in a field?

Reply:
Interesting idea actually. At the time this was written, I did not realize such a possibility on one hand, and probably wanted the changes to not metastacize all over the codebase on another.